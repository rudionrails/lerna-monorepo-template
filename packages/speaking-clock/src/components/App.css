/*
 * Widgets use CSS Modules to scope theyr styles
 */
.app :global(.material-icons) {
  font-size: inherit;
}

.app {
  min-height: 10rem;
}
