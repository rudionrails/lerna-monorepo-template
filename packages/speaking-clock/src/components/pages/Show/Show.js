import React from "react";
import PropTypes from "prop-types";

import Hours from "../../atoms/Hours";
import Minutes from "../../atoms/Minutes";

const Show = ({ date }) => {
  const hours = date.getHours();
  const minutes = date.getMinutes();

  return (
    <div className="text-center">
      <div className="small text-muted">the time is</div>

      <Hours {...{ hours, minutes }} />
      <Minutes {...{ minutes }} />
    </div>
  );
};

Show.propTypes = {
  date: PropTypes.shape({
    getHours: PropTypes.func.isRequired,
    getMinutes: PropTypes.func.isRequired,
  }).isRequired,
};

export default Show;
