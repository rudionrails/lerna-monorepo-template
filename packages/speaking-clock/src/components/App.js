import React from "react";

import Show from "./pages/Show";
import style from "./App.css";

const App = () => (
  <div className={`d-flex flex-column justify-content-center ${style.app}`}>
    <Show />
  </div>
);

export default App;
