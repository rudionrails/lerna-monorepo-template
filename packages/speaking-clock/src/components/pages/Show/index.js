import { connect } from "../../../store";
import Show from "./Show";

export default connect(state => ({
  date: state.date,
}))(Show);
