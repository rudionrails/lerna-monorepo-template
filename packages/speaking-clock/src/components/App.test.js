import React from "react";

import Show from "./pages/Show";

// module under test
import App from "./App";

test("to render Show", () => {
  const subject = shallow(<App />).find(Show);
  expect(subject.exists()).toBe(true);
});

test("to match snapshot", () => {
  const subject = shallow(<App />);
  expect(subject).toMatchSnapshot();
});
