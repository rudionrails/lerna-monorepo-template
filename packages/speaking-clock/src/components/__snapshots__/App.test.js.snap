// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  className="d-flex flex-column justify-content-center app"
>
  <Component />
</div>
`;
